experiment/tracer/static/script.js

```diff
@@ -0,0 +1,273 @@
+'use strict';
+
+var logger = {
+  data: [],
+  message: '',
+  options: {
+    request: {
+      endpoint: 'https://' + window.location.hostname + '/trace?',
+      params: '',
+      config: {}
+    },
+    messages: {
+      invalidInput: 'Invalid input. Please pass a link from a PR or PR comment and profit',
+      invalidHost: 'Invalid host. Host needs to be github.com',
+      invalidParams: 'Invalid params in the URL. Need either "pr", "repo", and "org", or  "issuecomment" to be specified',
+      invalidUrl: 'Invalid link provided. Failed to construct \'URL\'',
+      requestError: 'Fetch error: Status is not 2xx',
+      requestLoading: 'Loading..',
+      emptyLogs: 'No logs to display'
+    }
+  },
+
+  // the actual fetch
+  request: function(endpoint) {
+    generateView.toggleLoader(true);
+    fetch(endpoint).then(
+        function(response) {
+
+          if (response.status !== 200) {
+            generateView.toggleLoader(false);
+            controller.setMessage(logger.options.messages.requestError);
+            return;
+          }
+
+          response.json().then(function(data) {
+            generateView.toggleLoader(false);
```

Comment:
can chain `catch(...).finally(() => {generateView.toggleLoader(false);})` and remove all other toggleLoader